### PR TITLE
Test-related fixes (RLTest updates and more)

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: tests/pytests/test_config.py [hunk 1]
```diff
@@ -153,14 +154,14 @@ def test_arg_true(arg_name):
         env.stop()
 
     test_arg_true('NOGC')
-    test_arg_true('SAFEMODE')
+    # test_arg_true('SAFEMODE')
```
Comment (MeirShpilraien) [inline tests/pytests/test_config.py:157]: Why was this deleted?

File: tests/pytests/test_profile.py [hunk 2]
```diff
@@ -97,12 +97,12 @@ def testProfileSearch(env):
                     ['Type', 'INTERSECT', 'Counter', 1L, 'Children iterators',
                       ['Type', 'TEXT', 'Term', 'hello', 'Counter', 1L, 'Size', 1L],
                       ['Type', 'INTERSECT', 'Counter', 1L, 'Children iterators',
-                        ['Type', 'TEXT', 'Term', 'hello', 'Counter', 1L, 'Size', 1L]
+                        ['Type', 'TEXT', 'Term', 'hello', 'Counter', 1L, 'Size', 1L],
```
Comment (MeirShpilraien) [inline tests/pytests/test_profile.py:100]: How did it worked without it? @ashtul FYI

**Review by MeirShpilraien — COMMENTED** [review]
:+1: 
2 small questions

File: tests/pytests/test_profile.py (hunk 2 above)
Comment (rafie) [reply tests/pytests/test_profile.py:100]: It did not.

File: tests/pytests/test_config.py (hunk 1 above)
Comment (rafie) [reply tests/pytests/test_config.py:157]: @MeirShpilraien This test will fail when run with `CONCURRENT_WRITE_MODE`. Simply not a good test anymore, once `SAFEMODE` was subordinated to `CONCURRENT_WRITE_MODE`.

File: tests/pytests/test_profile.py (hunk 2 above)
Comment (ashtul) [reply tests/pytests/test_profile.py:100]: This fix was pushed while RLtest was already broken

File: tests/pytests/test_config.py (hunk 1 above)
Comment (MeirShpilraien) [reply tests/pytests/test_config.py:157]: :+1:

**Review by MeirShpilraien — DISMISSED** [review]

**Review by ashtul — APPROVED** [review]
